### ISSUE TITLE
Cosine annealing learning rate scheduler with minimum learning rate boundary

### ISSUE THREAD

**Issue by hfahrudin** [tensorflow/tensorflow#47493]
**System information**
- TensorFlow version (you are using): Tensorflow 2.x
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
In the nightly build, we have API called [tf.keras.optimizers.schedules.CosineDecay](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/CosineDecay)  which schedule decaying learning rate. However, we cannot set the range of learning rate incremental decay. For example, this current API allow us to change learning from alpha then decrease it in cosine manner and beyond. My proposal is to decrease the learning rate incrementally in cosine manner from alpha to beta.
note - alpha : initial learning rate , beta : final learning rate

**Will this change the current api? How?**
It could be implemented in tf.keras.optimizers.schedules.CosineDecay or implemented in different API name.

**Who will benefit with this feature?**
Anyone who want to use cosine annealing learning rate and worry about vanishing gradient issue.

**Comment by hfahrudin** [comment]
Could i start working on this?

**Comment by jvishnuvardhan** [comment]
@hfahrudin Sure. Thanks for your contributions. You can raise a PR to update it. Thanks!

**Comment by hfahrudin** [comment]
@jvishnuvardhan Should i make another class, or just edit this [API](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/CosineDecay)?

**Comment by deeb02** [comment]
@hfahrudin I think it's sensible to add this as an optional argument to the current API.

**Comment by hfahrudin** [comment]
I already issued the pull request but it seems that i issued it to protected branch. Which branch should i use for this PR?

**Comment by mihaimaruseac** [comment]
Always master branch

**Comment by deeb02** [comment]
Hi @hfahrudin,
I took a look at the PR you submitted. Upon looking again at the math, it does seem that the existing implementation provides a minimum/final learning rate, but it's defined as a fraction of the initial learning rate (not as the actual final learning rate).
Another point that struck me, is that your PR changes the calculation (whereas I initially though it would just add an optional minimum), this makes it backwards incompatible and makes me very hesitent to accept it unless it actually fixes a bug.

**Comment by hfahrudin** [comment]
@deeb02 Thanks for the review. I would keep it in mind in the future

**Comment by deeb02** [comment]
Absolutely. Thank you for trying to make Keras better!
Keep looking for things that need to be improved!
